### PR TITLE
Zero based depositID, docstrings for DepositManager.sol

### PR DESCRIPTION
**Pull request by jacque006**
Resolves https://github.com/thehubbleproject/hubble-contracts/issues/642

### CODE REVIEW

**Comment by jacque006** [comment]
Gas changes, very minor

### master 

```
Deposit 0 transaction cost 142214
Deposit 1 transaction cost 110336
Deposit 2 transaction cost 108026
```

### This PR

```
Deposit 0 transaction cost 142240
Deposit 1 transaction cost 110361
Deposit 2 transaction cost 108052
```

**Review by kautukkundan — APPROVED** [review]
doc-strings are a great addition!

File: contracts/DepositManager.sol [hunk 1]
```diff
@@ -10,26 +10,55 @@ import { ImmutableOwnable } from "./libs/ImmutableOwnable.sol";
...
         uint256 pubkeyID,
         uint256 tokenID,
         uint256 l2Amount,
         uint256 subtreeID,
         uint256 depositID
     );
+    /**
+     * @notice Event when a deposit subtree is ready
+     * to be submitted to the rollup
+     * @param subtreeID Subtree ID of deposits (1 - n)
```
Comment (ChihChengLiang) [inline contracts/DepositManager.sol:36]:
```suggestion
     * @param subtreeID Subtree ID of deposits (1 ... n)
```

File: contracts/DepositManager.sol [hunk 2]
```diff
@@ -10,26 +10,55 @@ import { ImmutableOwnable } from "./libs/ImmutableOwnable.sol";
...
+     * @param subtreeRoot Merkilized root of subtree
```
Comment (ChihChengLiang) [inline contracts/DepositManager.sol:37]: Nitpick: There are different variations of spellings of "Merklized", but this looks new
```suggestion
     * @param subtreeRoot Merklized root of subtree
```

File: contracts/DepositManager.sol [hunk 3]
```diff
@@ -10,26 +10,55 @@ import { ImmutableOwnable } from "./libs/ImmutableOwnable.sol";
...
+    /**
+     * @notice Dequeues a deposit subtree for submission to the rollup
+     */
     function dequeueToSubmit()
         external
         returns (uint256 subtreeID, bytes32 subtreeRoot);
 
+    /**
+     * @notice Re-enqueues a deposit subtree.
+     * @param subtreeRoot Merkilized root of subtree
```
Comment (ChihChengLiang) [inline contracts/DepositManager.sol:55]:
```suggestion
     * @param subtreeRoot Merklized root of subtree
```

**Review by ChihChengLiang — COMMENTED** [review]
Core logic looks good. I added some nitpicks on docstrings.

**Pull request merged**